### DISCUSSION TITLE
How to add accessibility (specifically arrow keys) for custom components

### DISCUSSION THREAD

**Discussion by JoeVanGundy** [chakra-ui/chakra-ui#5867] [Q&A]
I've created my own Menu component.
Basically something like:
```
<VStack>
   <Box _focus={{bg: 'blue'}}>MenuItem 1</Box>
   <Box  _focus={{bg: 'blue'}}>MenuItem 2</Box>
   <Box _focus={{bg: 'blue'}}>MenuItem 2</Box>
</VStack>
```

Is there a way I can easily add keyboard functionality so that I can either tab or use the array keys to navigate through each item?

**Comment by RunDevelopment** [comment]
Tab support can be added using the `tabIndex` prop (see the [`tabindex` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) for more information).

**Comment by TylerAPfledderer** [comment]
Because it's also created by the creator of Chakra, how about try ZagJS? He provides state machine logic for menus which also includes the needed accessibility.

https://zagjs.com/components/react/menu
